### PR TITLE
Manning book proposal

### PR DESCRIPTION
**Pull request by gvwilson**
Please add references to other material, more detail in all sections, and your info.

### CODE REVIEW

File: manning/manning-proposal-2020-06.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+1.  Create a Unix shell script to re-run a series of data analysis steps on a new input file with a single command.
+2.  Use branches, pull requests, and code review to add a feature to the Python programs invoked by that script.
+3.  Create a Makefile that automatically regenerates results and plots whenever a new data file is added to the project.
+4.  Add error handling and logging to the scripts so that they can be put into production.
+5.  Use style-checking tools to ensure the Python code is well written, and then release it as an installable package.
+6.  Adopt a license, a code of conduct, and efficient decision-making processes to ensure the project runs smoothly.
+7.  Publish results with DOIs, ORCIDs, and other metadata in a fully-reproducible way.
+
+**4. The minimally-qualified reader (MQR)**
+
```
Comment (DamienIrving) [inline manning/manning-proposal-2020-06.md:81]:
```suggestion
At the beginning of the book we introduce three "characters" that the narrative centers around. Their profiles are as follows:

```

File: manning/manning-proposal-2020-06.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+All of our chapters are driven by questions we have answered (some of them many times)
+in workshops or in discussions with colleagues.
+Our answers for technical subjects frequently draw on Stack Overflow,
+particularly for topics like continuous integration and packaging;
+our answers for community management and project leadership draw on extensive experience in open source.
+
+**6. Tell us about the competition and the ecosystem.**
+
+*What other books are available on this topic?*
```
Comment (DamienIrving) [inline manning/manning-proposal-2020-06.md:128]: There's [A Tour of Data Science](https://www.routledge.com/A-Tour-of-Data-Science-Learn-R-and-Python-in-Parallel/Zhang/p/book/9780367895860), which from what I  can tell is based on an openly available resource called [Another Book on Data Science](https://www.anotherbookondatascience.com/index.html).

Comment (DamienIrving) [reply manning/manning-proposal-2020-06.md:128]: (A Tour of Data Science is set to be published in December 2020)

File: manning/manning-proposal-2020-06.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+*The Turing Way* (https://the-turing-way.netlify.app/), produced by the Turing Institute in the UK,
+is the only up-to-date resource we know of with similar breadth.
+Other books cover only one topic in more depth than our audience needs
+(e.g., Ray & Ray's *Unix and Linux: Visual QuickStart Guide*)
+or are aimed at commercial software developers
+(any number of books on testing, coding style, and dev ops).
+
+*How does the proposed book compare to them?*
```
Comment (DamienIrving) [inline manning/manning-proposal-2020-06.md:137]: Points of difference:
1. Our focus on programming best practices (e.g. principles like writing modular, reusable, testable code). Others teach basic Python and R syntax and how to complete isolated tasks with particular libraries, but in this day and age you can pick up that information from a Google search. Our best practice content is the bigger picture you can't glean from online code snippets or Stack Overflow answers.
2. We go through the process of building a data science workflow/package from scratch. Most other texts simply present information (like a reference book) - we tackle a real data science problem from beginning to end, weaving in relevant information as we go.
3. Our book is represents a ready to go university semester course. There is growing demand for Software Carpentry the semester course.

File: manning/manning-proposal-2020-06.md (hunk 1 above)
Comment (gvwilson) [reply manning/manning-proposal-2020-06.md:81]: :+1:

File: manning/manning-proposal-2020-06.md (hunk 3 above)
Comment (gvwilson) [reply manning/manning-proposal-2020-06.md:137]: :+1:

File: manning/manning-proposal-2020-06.md (hunk 2 above)
Comment (gvwilson) [reply manning/manning-proposal-2020-06.md:128]: :+1:

File: manning/manning-proposal-2020-06.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+-   For Damien Irving: irving.damien@gmail.com
+-   For Greg Wilson: gvwilson@third-bit.com
+
+Website/blogs/Twitter, etc:
+
+FIXME: PLEASE ADD YOURS
+
+-   For Damien Irving: https://damienirving.github.io/
+-   For Greg Wilson: http://third-bit.com
```
Comment (joelostblom) [inline manning/manning-proposal-2020-06.md:219]:
```suggestion
-   For Greg Wilson: http://third-bit.com
-   For Joel Ostblom: https://joelostblom.com 
```

File: manning/manning-proposal-2020-06.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+-   For Damien Irving: 124 Marlyn Road, South Hobart, Tasmania 7004, Australia
+-   For Greg Wilson: 65 Highfield Road, Toronto, Ontario M4L 2T9, Canada
+
+Preferred email:
+
+FIXME: PLEASE ADD YOURS
+
+-   For Damien Irving: irving.damien@gmail.com
+-   For Greg Wilson: gvwilson@third-bit.com
```
Comment (joelostblom) [inline manning/manning-proposal-2020-06.md:212]:
```suggestion
-   For Greg Wilson: gvwilson@third-bit.com
-   For Joel Ostblom: joel.ostblom@protonmail.com
```

File: manning/manning-proposal-2020-06.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+-   Damien Irving
+-   Greg Wilson
+
+Mailing Address:
+
+FIXME: PLEASE ADD YOURS
+
+-   For Damien Irving: 124 Marlyn Road, South Hobart, Tasmania 7004, Australia
+-   For Greg Wilson: 65 Highfield Road, Toronto, Ontario M4L 2T9, Canada
```
Comment (joelostblom) [inline manning/manning-proposal-2020-06.md:205]:
```suggestion
-   For Greg Wilson: 65 Highfield Road, Toronto, Ontario M4L 2T9, Canada
-   For Joel Ostblom: 4227 W 12th Ave, Vancouver, British Columbia, V6R 2P8, Canada
```

File: manning/manning-proposal-2020-06.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+-   Dr. Damien B. Irving
+-   Dr. Gregory V. Wilson
+
+Name (as it would appear on a book cover):
+
+FIXME: PLEASE ADD YOURS
+
+-   Damien Irving
+-   Greg Wilson
```
Comment (joelostblom) [inline manning/manning-proposal-2020-06.md:198]:
```suggestion
-   Greg Wilson
-   Joel Ostblom
```

File: manning/manning-proposal-2020-06.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+-   The approximate number of code listings: 150.
+
+**8. Contact information**
+
+Formal Name (as it should appear on a book contract):
+
+FIXME: PLEASE ADD YOURS
+
+-   Dr. Damien B. Irving
+-   Dr. Gregory V. Wilson
```
Comment (joelostblom) [inline manning/manning-proposal-2020-06.md:191]:
```suggestion
-   Dr. Gregory V. Wilson
-   Joel Ostblom
```

Comment (joelostblom) [reply manning/manning-proposal-2020-06.md:191]: Can hopefully prefix a fancy "Dr" in 2 months!

Comment (joelostblom) [reply manning/manning-proposal-2020-06.md:191]: Ahhh two months so little time!!!

File: manning/manning-proposal-2020-06.md [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+-   Joel Ostblom
```
Comment (lwjohnst86) [inline manning/manning-proposal-2020-06.md:192]:
```suggestion
-   Joel Ostblom
-   Dr. Luke W. Johnston
```

**Pull request closed without merging**